Added support for zone management

Yannic-HAW opened the pull request:
Hi,

i have added support for ZoneManagement to the tchaikovsky lib.
Just call .createZone() one the desired speaker and pass a list of speaker objects to add to the zone. Afterwards the speaker object you called with .createZone() will become the zone master and all speakers in the passed list will become slaves. 

To revoke the group just call .releaseZone() on the master speaker. 

Code is tested and works just fine for me. Maybe it is a useful feature you would like to pull into your branch.

Yannic-HAW committed:
- git ignore updated

Yannic-HAW committed:
- support for zones added
- TODO: implement Zone onChange callback

dominicdesu commented:
Awesome, thank you so much for contributing your implementation of the zone management!!
I will review the changes asap and come back to you in case of any questions.

dominicdesu commented on src/main/java/de/kaizencode/tchaikovsky/speaker/ZoneManager.java:
```diff
@@ -0,0 +1,52 @@
+     * @throws SpeakerException
+     *             if the group could not be created
+     */
+    public RemoteZoneItem createZone(String[] speakers) throws SpeakerException;
```
Could you return the ZoneItem interface here instead of the implementation?

dominicdesu commented on src/main/java/de/kaizencode/tchaikovsky/speaker/remote/RemoteZoneManager.java:
```diff
@@ -0,0 +1,64 @@
+    }
+
+    @Override
+    public RemoteZoneItem createZone(String[] speakers) throws SpeakerException {
```
Could you return the ZoneItem interface here instead of the implementation?

dominicdesu commented on src/main/java/de/kaizencode/tchaikovsky/speaker/remote/RemoteSpeaker.java:
```diff
@@ -131,8 +135,24 @@ public void setSessionTimeout(int timeoutInSec) {
     }
+    
+    @Override
+    public void createZone(List<Speaker> speakerItems) throws SpeakerException {
```
What do you think about putting createZone and releaseZone into the RemoteZoneManager class and just returning the ZoneManager interface instead? (Similar as done with the Volume interface). This way, users have access to all methods of the ZoneManager.

dominicdesu commented on src/main/java/de/kaizencode/tchaikovsky/speaker/remote/RemoteZoneManager.java:
```diff
@@ -0,0 +1,64 @@
+
+public class RemoteZoneManager implements ZoneManager {
+    
+    private final Logger logger = LoggerFactory.getLogger(RemoteVolume.class);
```
I guess this should be RemoteZoneManager.class?

dominicdesu reviewed (changes requested):
Hi @Yannic-HAW, sorry for the late review, I was very busy the last days. I added some small comments, could you check if it makes sense to change it before merging?

Yannic-HAW committed:
- implemented requested changes
- implemented onZoneChange callbacks
- implemented missing zonInfo items

Yannic-HAW commented:
Hi,
i have implemented your suggested changes and have also implemented the onZoneChange callback.
You had some code disabled in ZoneInfo interface and i think you did that because of the messed up returned Variant by the leadPlayerName parameter. 
I have changed that part a bit and added isLeadPlayer() and getLeadPlayerID() methods. 

I dont know if it just works for the panasonic models but with panasonic it works just fine that way.

dominicdesu reviewed (approved):
Thanks!
